controllers/userController.js

```diff
@@ -18,6 +18,18 @@ exports.getAllUsers = async (req, res) => {
 
 exports.updateUser = async (req, res) => {
   try {
+    const idToSearchFor = { _id: req.user['user:id'] }
```

Comment:
Aici trebuie sa face un convert de la străin la ObjectId. Este-o funcție din mongoose cu care poți face asta